test/unit/cmap/handshake/client_metadata.test.ts

```diff
@@ -1,149 +1,523 @@
+            version: os.release()
+          },
+          platform: `Node.js ${process.version}, ${os.endianness()}`
+        });
+      });
+    });
 
-    it('does not append the driver info to the metadata', () => {
-      expect(metadata).to.deep.equal({
-        driver: {
-          name: 'nodejs',
-          version: NODE_DRIVER_VERSION
-        },
-        os: {
-          type: os.type(),
-          name: process.platform,
-          architecture: process.arch,
-          version: os.release()
-        },
-        platform: `Node.js ${process.version}, ${os.endianness()}`
+    context('when driverInfo.platform is provided', () => {
+      it('appends driverInfo.platform to the platform field', () => {
+        const options = {
+          driverInfo: { platform: 'myPlatform' }
+        };
+        const metadata = makeClientMetadata(options);
+        expect(metadata).to.deep.equal({
+          driver: {
+            name: 'nodejs',
+            version: NODE_DRIVER_VERSION
+          },
+          os: {
+            type: os.type(),
+            name: process.platform,
+            architecture: process.arch,
+            version: os.release()
+          },
+          platform: `Node.js ${process.version}, ${os.endianness()}|myPlatform`
+        });
       });
     });
 
-    it('does not set the application field', () => {
-      expect(metadata).not.to.have.property('application');
+    context('when driverInfo.name is provided', () => {
+      it('appends driverInfo.name to the driver.name field', () => {
+        const options = {
+          driverInfo: { name: 'myName' }
+        };
+        const metadata = makeClientMetadata(options);
+        expect(metadata).to.deep.equal({
+          driver: {
+            name: 'nodejs|myName',
+            version: NODE_DRIVER_VERSION
+          },
+          os: {
+            type: os.type(),
+            name: process.platform,
+            architecture: process.arch,
+            version: os.release()
+          },
+          platform: `Node.js ${process.version}, ${os.endianness()}`
+        });
+      });
     });
-  });
 
-  context('when app name is provided', () => {
-    context('when the app name is over 128 bytes', () => {
-      const longString = 'a'.repeat(300);
-      const options = {
-        appName: longString,
-        driverInfo: {}
-      };
-      const metadata = makeClientMetadata(options);
-
-      it('truncates the application name to <=128 bytes', () => {
-        expect(metadata.application?.name).to.be.a('string');
-        // the above assertion fails if `metadata.application?.name` is undefined, so
-        // we can safely assert that it exists
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        expect(Buffer.byteLength(metadata.application!.name, 'utf8')).to.equal(128);
+    context('when driverInfo.version is provided', () => {
+      it('appends driverInfo.version to the version field', () => {
```

Comment:
case for exceeding max size?

Reply:
the throwing tests now moved up here cover this